src/zephyr/tool/get-projects.ts

```diff
@@ -0,0 +1,40 @@
+import type { ToolCallback } from "@modelcontextprotocol/sdk/server/mcp.js";
+import type { ApiClient } from "../common/api-client.js";
+import { MaxResultsSchema, StartAtSchema } from "../common/types.js";
+import type { ZephyrTool } from "./zephyr-tool.js";
+
+export class GetProjects implements ZephyrTool {
+  private readonly apiClient: ApiClient;
+
+  constructor(apiClient: ApiClient) {
+    this.apiClient = apiClient;
+  }
+
+  specification = {
+    title: "Get Projects",
+    summary: "Get details of projects in Zephyr",
+    parameters: [
+      {
+        name: "maxResults",
+        type: MaxResultsSchema,
+        required: false,
+      },
+      {
+        name: "startAt",
+        type: StartAtSchema,
+        required: false,
+      },
+    ],
+  };
+
+  handle: ToolCallback<{ [key: string]: any }> = async (args, _extra) => {
```

Comment:
The Pact team have a nice trick [here](https://github.com/SmartBear/smartbear-mcp/blob/3e87ebfa74cb917e8e1dfd35f755e68e1b28cc17/src/pactflow/client/base.ts#L128) to infer the type from the Zod schema – possibly something to use here so you have matching types in the callback.